src/main/java/org/springframework/metrics/instrument/Measurement.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.metrics.instrument;
+
+import java.util.*;
+
+/**
+ * A measurement sampled from a meter.
+ *
+ * @author Clint Checketts
+ * @author Jon Schneider
+ */
+public final class Measurement {
+
+    private final String name;
+    private final SortedSet<Tag> tags = new TreeSet<>(Comparator.comparing(Tag::getKey));
+    private final double    value;
```

Comment:
I'm trying to keep `MeterId` internal now for a few reasons:

1. I think it winds up creating a less friendly API for the end user. Spectator does this and
`registry.gauge(registry.createId(name, tags), ...).` just doesn't seem as nice as
`registry.gauge(name, tags, ...)`.

2. There are certain contexts where name and tags occur together and the tags need to be sorted (like here), and some where they do not.

3. I can't think of a good name for it. In the context of a measurement, it's really a different type of id - a name that is potentially prefixed by the containing meter's name, and a set of tags that may be a superset of the meter's tags.